ledger/store/src/transaction/deployment.rs

```diff
@@ -186,7 +197,9 @@ pub trait DeploymentStorage<N: Network>: Clone + Send + Sync {
         atomic_batch_scope!(self, {
             // Store the program ID.
             self.id_map().insert(*transaction_id, program_id)?;
-            // Store the edition.
+            // Store the edition for the transaction ID.
+            self.id_edition_map().insert(*transaction_id, edition)?;
```

Comment:
This will insert a 0 edition for new programs, but have a missing item for existing programs